src/main/kotlin/org/rust/ide/refactoring/introduceConstant/RsIntroduceConstantHandler.kt

```diff
@@ -75,11 +75,23 @@ private fun RsExpr.isExtractable(): Boolean {
     }
 }
 
-private fun replaceWithConstant(expr: RsExpr, occurrences: List<RsExpr>, candidate: InsertionCandidate, editor: Editor) {
+private fun replaceWithConstant(
+    expr: RsExpr,
+    occurrences: List<RsExpr>,
+    candidate: InsertionCandidate,
+    editor: Editor
+) {
     val project = expr.project
     val factory = RsPsiFactory(project)
     val suggestedNames = expr.suggestedNames()
-    val name = suggestedNames.default.toUpperCase()
+
+    val owner = candidate.parent
+    val bindings = (owner.children.first() as? RsElement)?.getAllVisibleBindings().orEmpty() +
```

Comment:
Btw `getAllVisibleBindings` uses resolve which triggers similar exception to #8043

Reply:
Hmm. What can we do about it?

Reply:
Maybe call `getAllVisibleBindings` in `RsIntroduceConstantHandlerKt#extractExpression`

Reply:
Hmm, but at that point I don't know the insertion point yet. Maybe a more general solution would be to just use something like `invokeLater { replaceWithConstant }`?

Reply:
Good idea, let's try `invokeLater` or `nonBlocking` (introduced in #7899)